mlflow/utils/search_utils.py

```diff
@@ -74,8 +73,10 @@ def _get_identifier(cls, identifier):
     def _process_token(cls, token):
         if token.ttype == TokenType.Operator.Comparison:
             return {"comparator": token.value}
-        elif token.ttype in cls.VALUE_TYPES:
+        elif token.ttype in cls.NUMERIC_VALUE_TYPES:
             return {"value": token.value}
+        elif token.ttype in cls.STRING_VALUE_TYPES:
+            return {"value": token.value.strip("'")}  # strip quotes
```

Comment:
Is this robust to something like `params.apple = '\\''`

Reply:
Legit. Lemme fix this in a few ways

- Based on identifier type (param vs metric i.e.) assert a particular type of RHS
-- params => string. detect type of quotes and trim them manually
-- metrics => numerics. cast the appropriately here
- Better error handling for wrong type of arguments